redisinsight/api/src/modules/database-recommendation/providers/database-recommendation.provider.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { DatabaseRecommendationEntity }
+  from 'src/modules/database-recommendation/entities/database-recommendation.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { plainToClass } from 'class-transformer';
+import { RecommendationDto, RecommendationsDto } from 'src/modules/database-recommendation/dto';
+import { ClientMetadata } from 'src/common/models';
+
+@Injectable()
+export class DatabaseRecommendationProvider {
+  private readonly logger = new Logger('DatabaseRecommendationProvider');
+
+  constructor(
+    @InjectRepository(DatabaseRecommendationEntity)
+    private readonly repository: Repository<DatabaseRecommendationEntity>,
+  ) {}
+
+  /**
+   * Save entire entity
+   * @param databaseId
+   * @param recommendationName
+   */
+  async create(databaseId: string, recommendationName: string): Promise<RecommendationDto> {
+    this.logger.log('Creating database recommendation');
+    return await this.repository.save(
+      plainToClass(DatabaseRecommendationEntity, { databaseId, name: recommendationName }),
+    );
+  }
+
+  /**
+   * Return list of database recommendations
+   * @param clientMetadata
+   */
+  async list(clientMetadata: ClientMetadata): Promise<RecommendationsDto> {
+    this.logger.log('Getting database recommendations list');
+    const recommendations = await this.repository
+      .createQueryBuilder('r')
+      .where({ databaseId: clientMetadata.databaseId })
```

Comment:
I'm not sure about this filter. I forgot requirements but please double check if we need to return disabled recommendations, for example